src/Pos/Wallet/Web/State/Storage.hs

```diff
@@ -182,18 +190,26 @@ getWalletAddresses =
 getAccountWAddresses :: AddressLookupMode
                   -> AccountId
                   -> Query (Maybe [CWAddressMeta])
-getAccountWAddresses mode accId = do
-    let fetch which = toList <<$>> preview (wsAccountInfos . ix accId . which)
-    withAccLookupMode mode (fetch aiAccounts) (fetch aiRemovedAccounts)
+getAccountWAddresses mode accId =
+    withAccLookupMode mode (fetch aiAddresses) (fetch aiRemovedAddresses)
+  where
+    fetch :: MonadReader WalletStorage m => Lens' AccountInfo CAddresses -> m (Maybe [CWAddressMeta])
+    fetch which = do
+        -- `preview` operates within a `MonadReader m` and applies the lens
```

Comment:
Not here probably, but it would be nice to have this explanation _somewhere_, or else someone inexperienced with `Lens` (like me) is going to waste a lot of time (like me) figuring out what these functions do from their types.